lib/elixir/lib/calendar/date_range.ex

```diff
@@ -16,98 +17,93 @@ defmodule Date.Range do
           first: Date.t(),
           last: Date.t(),
           first_in_iso_days: iso_days(),
-          last_in_iso_days: iso_days()
+          last_in_iso_days: iso_days(),
+          step: pos_integer | neg_integer
         }
 
   @typep iso_days() :: Calendar.iso_days()
 
-  defstruct [:first, :last, :first_in_iso_days, :last_in_iso_days]
+  defstruct [:first, :last, :first_in_iso_days, :last_in_iso_days, :step]
 
   defimpl Enumerable do
     def member?(%{first: %{calendar: calendar}} = range, %Date{calendar: calendar} = date) do
```

Comment:
This may be incorrect as you need to consider empty ranges. Remember to add empty range tests for all operations below. :)